Drucker-Prager wellbore with cohesion hardening (#2491)
GEOS-DEV/GEOS: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/dpWellbore/Example.rst

.. _ExampleDPWellbore:

#########################################################
Drucker-Prager Model with Hardening for Wellbore Problems
#########################################################

**Context**

> This is an alternative to the example :ref:`ExampleEDPWellbore` where the Drucker-Prager constitutive with cohesion hardening (see :ref:`DruckerPrager`) is considered. Analytical solutions to this problem are not provided from open literature but they can be derived following the guide of (Chen and Abousleiman 2017)  <https://www.sciencedirect.com/science/article/pii/S1365160917301090>`__. Details of those solutions are given in Python scripts associated to this example.

Reviewer: ```suggestion
This is an alternative to the tutorial example :ref:`ExampleEDPWellbore`, and the Drucker-Prager constitutive with cohesion hardening (see :ref:`DruckerPrager`) is hereby considered. Analytical solutions to this problem are not provided from literature work, however they can be derived following `(Chen and Abousleiman 2017)  <https://www.sciencedirect.com/science/article/pii/S1365160917301090>`__. Details of those solutions are given in Python scripts associated to this example.
```

Author: done

**Input file**

This example uses no external input files and everything required is
contained within two xml files that are located at:

.. code-block:: console

  inputFiles/solidMechanics/DruckerPragerWellbore_base.xml

.. code-block:: console

  inputFiles/solidMechanics/DruckerPragerWellbore_benchmark.xml

The related integrated test is

.. code-block:: console

  inputFiles/solidMechanics/DruckerPragerWellbore_smoke.xml

> The Drucker-Prager properties are specified in the ``Constitutive`` section: 

Reviewer: ```suggestion
The Drucker-Prager material properties are specified in the ``Constitutive`` section: 
```

Author: done

.. literalinclude:: ../../../../../../../inputFiles/solidMechanics/DruckerPragerWellbore_base.xml
    :language: xml
    :start-after: <!-- SPHINX_DP_WELLBORE_MATERIAL -->
    :end-before: <!-- SPHINX_DP_WELLBORE_MATERIAL_END -->

> Here, ``rock`` is designated as the material in the computational domain. Drucker Prager model ``DruckerPrager`` is used to simulate the elastoplastic behavior of ``rock``. The material parameters, ``defaultFrictionAngle``, ``defaultDilationAngle`` and ``defaultCohesion`` denote the friction angle, the dilation angle, and the cohesion, respectively. In this example, cohesion is increased by a linear hardening law that is defined by the parameter ``defaultHardeningRate``. The constitutive parameters such as the density, the bulk modulus, and the shear modulus are specified in the International System of Units.

Reviewer: ```suggestion
Here, ``rock`` is designated as the material in the computational domain. Drucker Prager model ``DruckerPrager`` is used to simulate the elastoplastic behavior of ``rock``. The material parameters, ``defaultFrictionAngle``, ``defaultDilationAngle`` and ``defaultCohesion`` denote the friction angle, the dilation angle, and the cohesion, respectively. In this example, the hardening of cohesion is described by a linear hardening law, which is governed by the parameter ``defaultHardeningRate``. The constitutive parameters such as the density, the bulk modulus, and the shear modulus are specified in the International System of Units.
```

Author: done

| :math:`\phi`     | Friction Angle          | [degree]         | 15.27         |
+------------------+-------------------------+------------------+---------------+
| :math:`\psi  `   | Dilation Angle          | [degree]         | 15.0         |
+------------------+-------------------------+------------------+---------------+
| :math:`h`        | Hardening Rate          | [MPa]            | 10.0          |
+------------------+-------------------------+------------------+---------------+
| :math:`\sigma_h` | Horizontal Stress       | [MPa]            | -11.25        |
+------------------+-------------------------+------------------+---------------+
| :math:`\sigma_v` | Vertical Stress         | [MPa]            | -15.0         |
+------------------+-------------------------+------------------+---------------+
| :math:`a_0`      | Initial Well Radius     | [m]              | 0.1           |
+------------------+-------------------------+------------------+---------------+
| :math:`p_w`      | Mud Pressure            | [MPa]            | -2.0          |
+------------------+-------------------------+------------------+---------------+

> The validations of GEOS results against analyticla results is shown in the figure below:

Reviewer 2: ```suggestion
The validation of GEOS results against analytical results is shown in the figure below:
```

Author: Done
